### PR TITLE
GG-31173: ODBC nullability info for columns

### PR DESCRIPTION
**Pull request by isapego**

### CODE REVIEW

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
     /** Scale. */
     public final int scale;
 
-    /** Client version. */
-    private final ClientListenerProtocolVersion ver;
+    /** Nullability. */
+    public final int nullability;
```
[the remark this answers is not in the document]
Comment (isapego) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java:49]: 1. I'm not sure that `columnNullableUnknown` is not used and can never be returned. Do you propose to use some kind of enum here? Or should I add a comment?
2. You are right. I'm not sure, why this and other fields are public. Will change.

[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java]: 1. This field is initialized only in one place in `OdbcRequestHandler`: `prop.notNull() ? columnNoNulls : columnNullable`. 
 `columnNullableUnknown` is never used if I'm not missing anything.

[the remark this answers is not in the document]
Comment (isapego) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java]: That's not the only place where nullability is taken from - see second constructor, it takes nullability from H2 result set column information.

[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java]: Ok then, let's keep it as is, but add a comment with a reference to `ResultSetMetaData` to indicate the meaning.

[the remark this answers is not in the document]
Comment (isapego) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/odbc/OdbcColumnMeta.java]: Added comment that I hope will make it more clear.

**Pull request merged**